### PR TITLE
Fix connection APIs for local provider

### PR DESCRIPTION
**Pull request by Yashsharma1911**
**Notes for Reviewers**

This PR fixes #

I was getting issues on last, PR that's why I created this new PR with new changes

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: server/models/connection_persister.go
Comment (Yashsharma1911) [inline server/models/connection_persister.go:57]: Just moved this to separate file, I'll replace this with utility

**Comment by Yashsharma1911** [comment]
https://github.com/meshery/meshery/assets/71271069/c0238e9d-a1cb-4b25-b5a1-20e466740916

File: server/models/default_local_provider.go
Comment (leecalcote) [inline server/models/default_local_provider.go:1096]: Are we being schema-driven here? No schema reference and specific version?

File: server/models/default_local_provider.go
Comment (leecalcote) [inline server/models/default_local_provider.go:1027]: What schema version is this model?

File: server/models/default_local_provider.go
Comment (leecalcote) [inline server/models/default_local_provider.go:204]: Should this be a Connection, @MUzairS15?

File: server/helpers/utils/utils.go
Comment (leecalcote) [inline server/helpers/utils/utils.go:359]: Where are the new error codes with full descriptions?

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: server/models/default_local_provider.go
Comment (Yashsharma1911) [reply server/models/default_local_provider.go:1096]: No we are not, we need to define schema for connections, I have checked meshery/schemas repo, the schema which is present there is API schema for connection not the connection schema

File: server/helpers/utils/utils.go
Comment (Yashsharma1911) [reply server/helpers/utils/utils.go:359]: @leecalcote we don't need new error codes to be defined, most of them are already defined and used in connection handlers.

**Comment by Yashsharma1911** [comment]
This PR is done already, after approval we should be good to merge if there is no other review, I'll create an new issue to define connection schema and use it, I think currently we only using schema reference for models and components right @MUzairS15 ?

**Comment by MUzairS15** [comment]
Connection schema is ready defined, what else are you referring to?

File: server/models/default_local_provider.go
Comment (MUzairS15) [reply server/models/default_local_provider.go:204]: It is a connection, the connection specific info is stored here.

Comment (MUzairS15) [reply server/models/default_local_provider.go:204]: Because each connection has different schema/properties we store the info in this connection’s  property

**Comment by Yashsharma1911** [comment]
> Connection schema is ready defined, what else are you referring to?

I was looking at this file https://github.com/meshery/schemas/blob/master/schemas/constructs/v1alpha2/connection.yml, is this correct location to check connection schema? this is openAPI schema

**Comment by leecalcote** [comment]
@Yashsharma1911 @MUzairS15 we're in need of creating and releasing golang package(s) from this repo.

**Comment by leecalcote** [comment]
@Yashsharma1911 @MUzairS15 it's time - https://github.com/meshery/schemas/issues/43

**Comment by MUzairS15** [comment]
> > Connection schema is ready defined, what else are you referring to?
> 
> I was looking at this file https://github.com/meshery/schemas/blob/master/schemas/constructs/v1alpha2/connection.yml, is this correct location to check connection schema? this is openAPI schema

Yes, this is correct

**Comment by MUzairS15** [comment]
> @Yashsharma1911 @MUzairS15 we're in need of creating and releasing golang package(s) from this repo.

Yes.👍

**Comment by leecalcote** [comment]
Please fix merge conflicts.

**Comment by leecalcote** [comment]
> Please fix merge conflicts.

Please fix merge conflicts, @Yashsharma1911

**Comment by Yashsharma1911** [comment]
Merge conflicts resolved, for now I'm moving with this PR

**Pull request merged**
